cpp/tests/utilities/column_wrapper.hpp

```diff
@@ -169,7 +169,7 @@ rmm::device_buffer make_elements(InputIterator begin, InputIterator end)
   cudf::size_type size = std::distance(begin, end);
   thrust::host_vector<ElementTo> elements;
   elements.reserve(size);
-  fixed_width_type_converter<ElementFrom, ElementTo>{}(begin, end, elements.begin());
+  fixed_width_type_converter<ElementFrom, ElementTo>{}(begin, end, std::back_inserter(elements));
   return rmm::device_buffer{elements.data(), size * sizeof(ElementTo)};
```

Comment:
Sure would be nice if `fixed_width_type_converter` were just an algorithm rather than a functor that calls an algorithm. Then `make_elements` could be a 1-liner.  (Would also be a 1-liner if the converter was just a functor that transforms a single element and could be passed to `std::transform()`).

I think replacing fixed_width_type_converter with a factory function that, depending on the To and From types, returns a lambda that can be passed to std::transform would be nicer, simpler, and more flexible.

Reply:
Agreed, this problem didn't used to exist when `make_elements` just invoked the `host_vector` ctor that accepts an iterator range to copy. I think even better than a `std::transform` would just be to make a transform iterator that's passed to the iterator range ctor:
```
auto transformer = fixed_width_type_converter<ElementFrom, ElementTo>{};
auto transform_begin = thrust::make_transform_iterator(begin, transformer);
thrust::host_vector<ElementTo> elements( transform_begin, transform_begin + size); 
```
```

Reply:
I pushed a change for this but it was failing on `COPYING_TEST` locally with 
```
thrust::system::system_error'
what():  CUDA free failed: cudaErrorLaunchFailure: unspecified launch failure
```
But I have seen this before and it has just been a local issue. I spent a while trying to troubleshoot but decided to push and see what happens.

Reply:
I pushed a fix for `COPYING_TEST` but not sure if it is reasonable. Should you be able to call the range constructor for `fixed_width_column_wrapper` when the range is a `thrust::device_vector`? That seemed to be causing the issue.

Reply:
> Should you be able to call the range constructor for `fixed_width_column_wrapper` when the range is a `thrust::device_vector`

That was never an intended use case for `column_wrapper`. The iterators should be host accessible iterators. Technically speaking, `device_vector<T>::begin` returns an iterator that is accessible from host, but it incurs a D2H copy for every element.

Reply:
Not sure why I didn't notice it before but your comment made me realize all that was missing was the `CUDA_HOST_DEVICE_CALLABLE` on the function object call operators :+1: Thanks :pray:

Reply:
Does it need to be both host and device callable? In general we should aim to only use that when absolutely needed. If it's only called from the device, just use CUDA_DEVICE_CALLABLE.

Reply:
@codereport I think the usecase of constructing a `fixed_width_column_wrapper` from ` device_vector` is erroneous. The iterators should be host accessible only.

Reply:
Pushed the changes for host only (I think you had it the other way around @harrism).